compiler/src/dmd/expressionsem.d

```diff
@@ -14175,6 +14175,10 @@ Expression toBoolean(Expression exp, Scope* sc)
                               exp.toChars(), t.toChars());
                 return ErrorExp.get();
             }
+            if (exp.isArrayLiteralExp() || exp.isStringExp())
+            {
+                deprecation(exp.loc, "boolean evaluation of array/string literals is deprecated");
```

Comment:
I suggest "[...] array expressions and string literals [...]"

Reply:
Array expressions are still allowed in boolean context, see:
https://issues.dlang.org/show_bug.cgi?id=4733#c38

Reply:
I did not expect such a debate. My suggestion was firstly so that the `/` is not used, then secondly to prevent the confusion between "array {literal|static|slice}".

Reply:
maybe just repeat "literals" ?

"[...] array literals an string literals [...]"

Reply:
or the right message depending on the expression. That would work too, to prevent "array/string"

Reply:
just use the expression token to display "array" or "string"